libraries/extractor/src/main/java/androidx/media3/extractor/ts/AdtsReader.java

```diff
@@ -527,7 +557,76 @@ private void parseAdtsHeader() throws ParserException {
       sampleSize -= CRC_SIZE;
     }
 
-    setReadingSampleState(output, sampleDurationUs, 0, sampleSize);
+    if (!hasOutputFormat && sampleSize >= AAC_PCE_MIN_SIZE) {
+      // As sample can fit a PCE, try reading it.
+      setReadingAacPceState(output, sampleDurationUs, sampleSize);
+    } else {
+      setReadingSampleState(output, sampleDurationUs, 0, sampleSize);
+    }
+  }
+
+  @RequiresNonNull({"pendingOutputFormat", "pceBuffer"})
+  void readAacProgramConfigElement() {
+    if (pceBuffer.readBits(3) == 5 /* PCE tag */) {
+      pceBuffer.skipBits(10); // Element instance tag, profile, sample frequency index
+
+      int extraBits = 0; // Number of bits beyond amount to be parsed.
+      extraBits += pceBuffer.readBits(4) * 5; // Front channel elements
+      extraBits += pceBuffer.readBits(4) * 5; // Side channel elements
+      extraBits += pceBuffer.readBits(4) * 5; // Back channel elements
+      extraBits += pceBuffer.readBits(2) * 4; // LFE channel elements
+      extraBits += pceBuffer.readBits(3) * 4; // Data elements
+      extraBits += pceBuffer.readBits(4) * 5; // Coupling channel elements
+
+      if (pceBuffer.readBit()) {
+        pceBuffer.skipBits(4); // Mono mixdown
+      }
+
+      if (pceBuffer.readBit()) {
+        pceBuffer.skipBits(4); // Stereo mixdown
+      }
+
+      if (pceBuffer.readBit()) {
+        pceBuffer.skipBits(3); // Matrix mixdown
+      }
+
+      // Calculate total PCE size including initial PCE tag and a zero length comment.
+      int pceSize = (pceBuffer.getPosition() + extraBits + 7) / 8 + 1;
```

Comment:
Why do we assume a `zero length comment` here? As per the spec, I cannot see to find such an assumption.

Reply:
Well, we don't need to copy across the comment so there's no point buffering it up. If we had to copy the comment we'd need to increase `AAC_PCE_MAX_SIZE` from 49 to 305, which will increase the pceBuffer size.

Having said that, if we increase AAC_PCE_MAX_SIZE to 50 we'd get the byte representing the comment size. We could read that to more robustly check if the PCE fits into the sample in the below conditional. I'll make this change.